### PR TITLE
Parallelise CI jobs

### PR DESCRIPTION
**Pull request by danigo99**
## Context

Tech Debt ticket: https://linear.app/prismic/issue/SM-879/[tech-debt]-spike-1-day-aadev-the-ci-can-run-parallel-jobs

## The Solution

Separating each step of the build job into different jobs that run in parallel after the installation job.

This can still be improved by looking into parallelisation of cypress e2e test since these are taking up ~80% of the total runtime 😢 

## Checklist before requesting a review
- [x] I hereby declare my code ready for review.
- [ ] If it is a critical feature, I have added tests.
- [x] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

---

image

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -40,14 +34,59 @@ jobs:
...
+    steps: 
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/restore-cache
+      - uses: ./.github/actions/setup-node
+      - uses: ./.github/actions/build-essential
       - name: Audit
         run: lerna run audit
+
+  e2e-tests:
+    needs: install-deps
```
Comment (Heyrows) [inline .github/workflows/test.yml:83]: ❓ Do parallelised steps run on the same machine ?
I'm wondering if the dependencies would be installed for this job or any other jobs.

Comment (danigo99) [reply .github/workflows/test.yml:83]: No they don't, each job is on a separate runner. 
So first the install step happens, the node modules are cached, then all the other jobs start at the same time. In each of them the cache is restored, and the tests happen.

Comment (Heyrows) [reply .github/workflows/test.yml:83]: I see, so they don't actually have to install anything themselves, good 👍

**Review by Heyrows — APPROVED** [review]

**Comment by Heyrows** [comment]
Sounds good, can't wait to see how to make it even faster.

**Pull request merged**